### PR TITLE
system/utils: Refactor implementation of 'date' command

### PR DESCRIPTION
**Pull request by lokeshbv**
Implementation of 'date' command is refactored using 'strptime' library function.

Signed-off-by: Lokesh B V <lokesh.bv@partner.samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e2e2bb8620247cce82367b291d604c0a0a36fdc9] - Code Rule Check Result:
apps/system/utils/kdbg_date.c:72: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)

**Comment by seinfra** [comment]
Target : [a1b62ce43569eb6fecab68cce7b8391917c7e24f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a1b62ce43569eb6fecab68cce7b8391917c7e24f] - Code Rule Check (C++) OK.

File: apps/system/utils/README.md [hunk 1]
```diff
@@ -132,19 +132,19 @@ This command prints the date. This has an argument which sets system time and da
...
+               FORMAT: "MMM DD HH:MM:SS YYYY"
                 'month', 'day', 'hour':'minute':'second' 'year'
-                Example: Apr 21 10:35:22 1991
+                Example: "Apr 21 10:35:22 1991"
 
 TASH>>date
 Jan 01 00:00:09 2010
 
-TASH>>date -s Feb 27 14:41:05 2018
+TASH>>date -s "February 27 14:41:05 2018"
```
Comment (sunghan-chang) [inline apps/system/utils/README.md]: Both "Feb" and "February" are available?

File: apps/system/utils/kdbg_date.c [hunk 2]
```diff
@@ -67,6 +67,9 @@
 #define MAX_TIME_STRING         80
 #define STR_LEN_OF_MONTH_WEEK   3
 #define MAX_EPOCH_YEAR_OF_UINT  2106
+#define EPOCH_YEAR_CONST        70
```
Comment (sunghan-chang) [inline apps/system/utils/kdbg_date.c]: This is not fixed. This value is depends on TM_YEAR_BASE and EPOCH_YEAR.
Let's modify this like below
```
#define EPOCH_YEAR_CONST   (TM_YEAR_BASE - EPOCH_YEAR)

File: apps/system/utils/kdbg_date.c [hunk 3]
```diff
@@ -153,156 +155,40 @@ static inline int date_showtime(void)
 	return OK;
 }
 
-static inline int date_settime(int argc, char **args)
+static inline int date_settime(const char *tbuf)
 {
+#if defined(CONFIG_LIBC_LOCALTIME) || defined(CONFIG_TIME_EXTENDED)
+	const char format[] = "%b %d %H:%M:%S %Y %a %j";
```
Comment (sunghan-chang) [inline apps/system/utils/kdbg_date.c]: Usages says as shown below:
```
printf("               FORMAT: \"MMM DD HH:MM:SS YYYY DWR DYR D\"\n");
printf("                'month', 'day', 'hour':'minute':'second' 'year' 'day_of_week' 'day_of_year' 'daylight_savings_flag'\n");
printf("                Example: \"Apr 21 10:35:22 1991 Sun 175 1\"\n");
```
Where is the DST at format? Doesn't it support? Then we should modify Usages as well.

Comment (lokeshbv) [reply apps/system/utils/kdbg_date.c]: `strptime ` doesn't support DST, So the DST parsing logic is added here https://github.com/Samsung/TizenRT/blob/8fccc6ecf0d41ebabae4bb2229130ab3fef18bbf/apps/system/utils/kdbg_date.c#L182

**Comment by seinfra** [comment]
Target : [8fccc6ecf0d41ebabae4bb2229130ab3fef18bbf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8fccc6ecf0d41ebabae4bb2229130ab3fef18bbf] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@lokeshbv Our P&I lab tested the "date" command with yours. It replies differently against current codes.
```
TASH>>date -s "Jan 01 12:00:00 1969"
TASH>>date
Feb 06 18:28:18 2105
  => This should be invalid param. year should be in 1970~2106.

TASH>>date -s "Jan 32 12:00:00 2018"
TASH>>date
Jan 01 12:00:55 1970
  => This should be invalid param. Jan does not have 32.

TASH>>date -s "Jan 01 12:00:00 1923"
TASH>>date 
Feb 06 18:28:27 2059
  => same as first one.
```

File: apps/system/utils/kdbg_date.c (hunk 2 above)
Comment (lokeshbv) [reply apps/system/utils/kdbg_date.c]: `TM_YEAR_BASE -> 1900` and `EPOCH_YEAR -> 1970`, So the definition should be ```#define EPOCH_YEAR_CONST   (EPOCH_YEAR - TM_YEAR_BASE )```

@sunghan-chang find the expected result below,
```
TASH>>
TASH>>date -s "Jan 01 12:00:00 1969"
date: cannot set date: Invalid year in the argument!

TASH>>
TASH>>
TASH>>date -s "Jan 32 12:00:00 2018"
date: cannot set date: Invalid date-time argument!

TASH>>date -s "Jan 01 12:00:00 1923"
date: cannot set date: Invalid year in the argument!

TASH>>
```

**Comment by seinfra** [comment]
Target : [d43593943728cacfaa1c58a8c7e83be8a574a88e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d43593943728cacfaa1c58a8c7e83be8a574a88e] - Code Rule Check (C++) OK.

**Pull request closed without merging**